core/shared/src/main/scala/zio/logging/internal/FileWriter.scala

```diff
@@ -15,30 +15,75 @@
  */
 package zio.logging.internal
 
-import java.io.{ BufferedWriter, FileOutputStream, OutputStreamWriter, Writer }
+import zio.logging.FileLoggerConfig
+import zio.logging.FileLoggerConfig.FileRollingPolicy
+
+import java.io.{ BufferedWriter, File, FileOutputStream, OutputStreamWriter, Writer }
 import java.nio.charset.Charset
-import java.nio.file.Path
+import java.nio.file.{ FileSystems, Path }
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 
 private[logging] class FileWriter(
   destination: Path,
   charset: Charset,
   autoFlushBatchSize: Int,
-  bufferedIOSize: Option[Int]
+  bufferedIOSize: Option[Int],
+  rolling: Option[FileLoggerConfig.FileRollingPolicy]
 ) extends Writer {
-  private val writer: Writer = {
-    val output = new OutputStreamWriter(new FileOutputStream(destination.toFile, true), charset)
+  private var currentDestination             = destination
+  private def makeWriter(path: Path): Writer = {
+    val output = new OutputStreamWriter(new FileOutputStream(path.toFile, true), charset)
     bufferedIOSize match {
       case Some(bufferSize) => new BufferedWriter(output, bufferSize)
       case None             => output
     }
   }
+  private var writer: Writer                 = rolling match {
+    case Some(policy) =>
+      policy match {
+        case FileRollingPolicy.TimeBasedRollingPolicy =>
+          val newPath = makeDatePath()
+          currentDestination = newPath
+          makeWriter(newPath)
+      }
+    case None         => makeWriter(destination)
+  }
 
   private var entriesWritten: Long = 0
 
   final def write(buffer: Array[Char], offset: Int, length: Int): Unit =
     writer.write(buffer, offset, length)
 
+  private def makeDatePath(): Path = {
+    val currentDateTime = LocalDateTime.now()
+    val formatter       = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+    val time            = formatter.format(currentDateTime)
+    val fileNameArray   = destination.getFileName.toString.split("\\.")
+    val timeFileName    = if (fileNameArray.length >= 2) {
+      fileNameArray.dropRight(1).mkString(".") + "-" + time + "." + fileNameArray.last
+    } else {
+      fileNameArray.head + "-" + time
+    }
+    val timeDestination = FileSystems.getDefault.getPath(destination.getParent.toString, timeFileName)
+    timeDestination
+  }
+
   final def writeln(line: String): Unit = {
+    rolling match {
+      case Some(policy) =>
+        policy match {
+          case FileRollingPolicy.TimeBasedRollingPolicy =>
+            val newPath = makeDatePath()
+            if (newPath != currentDestination) {
```

Comment:
we will need more optimal comparison probably just based  on `currentDateTime`, 
as `makeDatePath` is relatively complex, to do it with every log line

we need to have logging as fast as possible, we need to avoid unnecessary memory allocations and computations